io/read_all.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+import { concat } from "../bytes/concat.ts";
+import { DEFAULT_CHUNK_SIZE } from "./_constants.ts";
+import type { Reader, ReaderSync } from "./types.d.ts";
+
+/**
+ * Read {@linkcode Reader} `r` until EOF (`null`) and resolve to the content as
+ * {@linkcode Uint8Array}.
+ *
+ * @example
+ * ```ts
+ * import { readAll } from "https://deno.land/std@$STD_VERSION/io/read_all.ts";
+ *
+ * // Example from stdin
+ * const stdinContent = await readAll(Deno.stdin);
+ *
+ * // Example from file
+ * const file = await Deno.open("my_file.txt", {read: true});
+ * const myFileContent = await readAll(file);
+ * file.close();
+ * ```
+ */
+export async function readAll(reader: Reader): Promise<Uint8Array> {
```

Comment:
This function differs from `readAll()` from `std/streams` in that it doesn't depend on `Buffer()` from `std/io`. It's simpler, only relying on `concat()`. This should be a non-breaking change.